src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Application {
+    public static void main(String[] args) {
+        int totalNumberOfFigures = 11;
+
+        Figure [] result = new Figure[totalNumberOfFigures];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < totalNumberOfFigures / 2; i++) {
+            result [i] = figureSupplier.getRandomFigure();
+            if (result [i] == null) {
+                System.out.println("Figure no " + i + "is null");
+            } else {
+                result[i].draw();
+            }
+        }
+
+        for (int i = totalNumberOfFigures / 2; i < totalNumberOfFigures; i++) {

```

Comment:
The code doesn't account for cases where totalNumberOfFigures is an odd number. This may lead to off-by-one errors.

Reply:
Why do you think so? 
In past version there was mechanizm if-else but you recommended to reduce it and I think you had right and now it's steel working correct.